kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/CredentialsValidator.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Confluent, Inc.
+ *
+ * This software contains code derived from the WePay BigQuery Kafka Connector, Copyright WePay, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.wepay.kafka.connect.bigquery.config;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.storage.Storage;
+import com.wepay.kafka.connect.bigquery.GcpClientBuilder;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig.ENABLE_BATCH_CONFIG;
+import static com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig.GCS_BUCKET_NAME_CONFIG;
+import static com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig.KEYFILE_CONFIG;
+import static com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig.KEY_SOURCE_CONFIG;
+import static com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig.PROJECT_CONFIG;
+
+public abstract class CredentialsValidator<ClientBuilder extends GcpClientBuilder<?>> extends MultiPropertyValidator<BigQuerySinkConfig> {
+
+  public CredentialsValidator() {
+    super(KEYFILE_CONFIG);
+  }
+
+  private static final Collection<String> DEPENDENTS = Collections.unmodifiableCollection(Arrays.asList(
+      PROJECT_CONFIG, KEY_SOURCE_CONFIG
+  ));
+
+  @Override
+  protected Collection<String> dependents() {
+    return DEPENDENTS;
+  }
+
+  @Override
+  protected Optional<String> doValidate(BigQuerySinkConfig config) {
+    String keyFile = config.getKey();
+    if (keyFile == null || keyFile.isEmpty()) {
+      // No credentials to validate
+      return Optional.empty();
+    }
+
+    try {
+      clientBuilder()
+          .withConfig(config)
+          .build();
+      return Optional.empty();
+    } catch (RuntimeException e) {
+      if (e.getMessage() == null) {
+        return Optional.of("An unexpected error occurred while validating credentials for " + gcpService());
```

Comment:
Do we need this?

Reply:
Yeah fair, I wanted to avoid having a trailing `": "` at the end if `e.getMessage()` was null but this is much more complicated than it needs to be just to handle that case.